File: architecture/clean-architecture/javascript/core/use-cases/Read.js
import UseCase from './UseCase.js';

export default class Read extends UseCase {
    constructor(repo, presenter) {
        super(repo, presenter);
    }
    execute(paramObj) {
        const { filter } = paramObj;
        const documents = this.repo.read(filter)
        return this.presenter.present(documents);
    }
}

File: architecture/clean-architecture/java/core/usecases/Read.java
package core.usecases;

import java.util.List;
import core.ports.IPresenter;
import core.ports.IRepository;
import core.entities.Document;
import core.entities.Response;
import core.entities.ParamObj;

public class Read {
    private final IRepository repo;
    private final IPresenter<List<Document>> presenter;

    public Read(IRepository repo, IPresenter<List<Document>> presenter) {
        this.repo = repo;
        this.presenter = presenter;
    }
    public final Response<List<Document>> execute(ParamObj params) {
        List<Document> documents = repo.read(params.filter);
        return presenter.present(documents);
    }
}
